source.go

```diff
@@ -82,15 +85,43 @@ func (s *Source) Config() sdk.SourceConfig {
 }
 
 func (s *Source) Open(ctx context.Context, pos opencdc.Position) error {
-	sdk.Logger(ctx).Info().Msg("Opening DynamoDB Source...")
-
-	cfg, err := config.LoadDefaultConfig(ctx,
-		config.WithRegion(s.config.AWSRegion),
-		config.WithCredentialsProvider(aws.NewCredentialsCache(credentials.NewStaticCredentialsProvider(s.config.AWSAccessKeyID, s.config.AWSSecretAccessKey, s.config.AWSSessionToken))),
-	)
-	if err != nil {
-		return fmt.Errorf("could not load AWS config: %w", err)
-	}
+    sdk.Logger(ctx).Info().Msg("Opening DynamoDB Source...")
+
+    // Build up LoadDefaultConfig options
+    opts := []func(*config.LoadOptions) error{
+        config.WithRegion(s.config.AWSRegion),
+    }
+
+    // If the user supplied static creds, register them first
+    if s.config.AWSAccessKeyID != "" && s.config.AWSSecretAccessKey != "" {
+        opts = append(opts,
+            config.WithCredentialsProvider(
+                aws.NewCredentialsCache(
+                    credentials.NewStaticCredentialsProvider(
+                        s.config.AWSAccessKeyID,
+                        s.config.AWSSecretAccessKey,
+                        s.config.AWSSessionToken,
+                    ),
+                ),
+            ),
+        )
+    }
+
+    // If the user also supplied a Role ARN, wrap the provider in an STS assume-role
+    if s.config.AWSRoleARN != "" {
+        opts = append(opts,
+            config.WithAssumeRoleCredentialOptions(func(o *stscreds.AssumeRoleOptions) {
+                o.RoleARN = s.config.AWSRoleARN
+                o.RoleSessionName = "conduit-dynamodb-connector"
```

Comment:
I'd make this also configurable through the config. Also, I'd change the default to `conduit-connector-dynamodb`, just so it exactly matches the connector name.